lib/pure/asyncmacro.nim

```diff
@@ -447,8 +325,28 @@ macro multisync*(prc: untyped): untyped =
   result = newStmtList()
   result.add(asyncSingleProc(asyncPrc))
   result.add(sync)
-
-proc await*[T](x: T) =
-  ## The 'await' keyword is also defined here for technical
-  ## reasons. (Generic symbol lookup prepass.)
-  {.error: "Await only available within .async".}
+  # echo result.repr
+
+# overload for type which is not Future[T], based on the yglukhov's patch to chronos: https://github.com/status-im/nim-chronos/pull/47
+template await*(f: untyped): untyped =
+  when declared(retFuture):
+    static:
+      error "await expects Future[T], got " & $typeof(f)
+  elif not declared(inMultisync):
+    static:
+      error "await only available within {.async.}"
+  else:
+    f
```

Comment:
From what I can tell this is only here to allow multisync to work. It took me a while to figure this out, can you add a comment to explain this?

Reply:
Actually, now that I look at the code above, I would just remove this and keep the `stripAwait` logic in `splitProc`. Adding this `inMultisync` variable into the generated code is more hacky. That way you can also reuse the compiler's type mismatch error messages.

Reply:
i really think this is also elegant: after all you change how this works quickly with high level code flow without any ast surgery. inMultisync should be somehow optimizeable, its needed only on compile time

Reply:
A loop which modifies the AST may be more code, but it's also far more deterministic and easier to debug should something go wrong. Also, when in doubt, leave things as they are.

Reply:
but .. wouldn't this break templates again? after all you wouldn't be able to generate await in multisync, i think

Reply:
oh, you mean if a template using `await` is used in a multisync proc? I think you're right. But in that case, can we resolve that in a separate PR? I just like to keep this clean.

Reply:
ok, but we really shouldn't get into a situation where just one of the cases is fixed.. they are symmetric

Reply:
ok, no we dont need those variables anyway, so i think this can be still one PR